lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -2568,6 +2568,97 @@ class ConvertAtenPermuteOp : public OpConversionPattern<AtenPermuteOp> {
 };
 } // namespace
 
+namespace {
+class ConvertAtenSliceTensorOp : public OpConversionPattern<AtenSliceTensorOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenSliceTensorOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+
+    Location loc = op.getLoc();
+    TypeConverter *typeConverter = getTypeConverter();
+
+    auto input = adaptor.self();
+    RankedTensorType inputType = input.getType().cast<RankedTensorType>();
+    RankedTensorType resultType =
+        typeConverter->convertType(op->getResult(0).getType())
+            .cast<RankedTensorType>();
+    int64_t resultRank = resultType.getRank();
+    SmallVector<Value> resultShape;
+    Value zero = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+    Value one = rewriter.create<arith::ConstantIndexOp>(loc, 1);
+
+    SmallVector<Value> offsets, strides;
+    strides.resize(inputType.getRank(), one);
+    offsets.resize(inputType.getRank(), zero);
+    Value end, start, stepIndex;
+    resultShape = getTensorSizes(rewriter, loc, input);
```

Comment:
I think the variable name `resultShape` is confusing here. `inputShape` would be better.

Reply:
May I respectfully disagree? Later on we are changing resultShape[dim] to the correct size for the resulting slice, hence "resultShape". We are obtaining the information from the input's shape, but not retaining that shape. Maybe I can add a comment above resultShape = getTensorSizes(rewriter, loc, input); // Initialized based on input shape

Reply:
ha, I see. Sry, I missed that. I think the confusion comes from reading the line below `auto temp = castIndexToInt(rewriter, loc, resultShape[dim]);`  It's surprising to the reader `resultShape` is used there. I would replace all the places currently referring to `resultShape[dim]` to a variable named `slicedDimSize`.

Reply:
Personally, I would use `Value slicedDimSize = getDimOp(...)` and do
```
resultShape = getTensorSizes(rewriter, loc, input);
resultShape[dim] = resultSize;
```
but that's up to you. 
Sry, after reading so many 20 years legacy code, I am paranoid about readability.

Reply:
Ok, that makes sense, I'll do it that way